core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt

```diff
@@ -10,50 +10,23 @@ import ch.ergon.dope.validtype.NumberType
 import ch.ergon.dope.validtype.StringType
 import ch.ergon.dope.validtype.ValidType
 
-class Primitive<T : ValidType> : TypeExpression<T> {
-    private val dopeQuery: DopeQuery
+open class Primitive<T : ValidType>() : TypeExpression<T> {
+    private var queryString: String = ""
 
-    override fun toDopeQuery(): DopeQuery = dopeQuery
-
-    constructor(value: Number) {
-        this.dopeQuery = DopeQuery(queryString = "$value", parameters = emptyMap())
-    }
-
-    constructor(value: String) {
-        this.dopeQuery = DopeQuery(queryString = "\"$value\"", parameters = emptyMap())
-    }
-
-    constructor(value: Boolean) {
-        this.dopeQuery = DopeQuery(
-            queryString = when (value) {
-                true -> TRUE.toDopeQuery().queryString
-                false -> FALSE.toDopeQuery().queryString
-            },
-            parameters = emptyMap(),
-        )
-    }
-
-    constructor(collection: Collection<TypeExpression<out ValidType>>) {
-        val dopeQueries = collection.map { it.toDopeQuery() }
-        this.dopeQuery = DopeQuery(
-            queryString = dopeQueries.joinToString(separator = ", ", prefix = "[", postfix = "]") { it.queryString },
-            parameters = dopeQueries.fold(emptyMap()) { parameters, dopeQueryElement ->
-                parameters + dopeQueryElement.parameters
-            },
-        )
-    }
-
-    private constructor(primitiveType: PrimitiveType) {
-        val queryString = when (primitiveType) {
+    constructor(type: PrimitiveType) : this() {
```

Comment:
Keep it Private. The companion Object contains all the Public instances for PrimitiveTypes.